test/framework/cluster.go

```diff
@@ -93,6 +95,10 @@ func NewClusterE2ETest(t *testing.T, provider Provider, opts ...ClusterE2ETestOp
 		opt(e)
 	}
 
+	e.T.Cleanup(func() {
```

Comment:
We should set this cleanup function once we know that everything on the provider is set up to create the VMs right? That would mean moving it a line below.

Reply:
done thanks!